### PR TITLE
VCS Integration explained

### PR DESCRIPTION
**Pull request by naster01**
Details for how to integrate with GitHub, Bitbucket or Gitlab.

### CODE REVIEW

File: docs/vcs.rst [hunk 1]
```diff
@@ -39,3 +45,30 @@ It can be used like this::
...
+      }
+
+It can be used like this::
+
+      {% if display_gitlab %}
+        <a href="https://{{ gitlab_host|default("gitlab.com") }}/{{ gitlab_user }}/{{ gitlab_repo }}/blob/{{ gitlab_version }}{{ conf_py_path }}{{ pagename }}{{ suffix }}" class="fa fa-gitlab"> Edit on GitLab</a>
+      {% endif %}
+
+
+Additionnal theme's variables:
```
Comment (berkerpeksag) [inline docs/vcs.rst]: Typo: Additionnal

Comment (berkerpeksag) [inline docs/vcs.rst]: Or we could just say: "Additional variables"

File: docs/vcs.rst [hunk 2]
```diff
@@ -39,3 +45,30 @@ It can be used like this::
...
+-------------------------------
+
+* pagename - Sphinx variable representing the name of the page you're on.
```
Comment (berkerpeksag) [inline docs/vcs.rst]: \`\`'pagename'\`\`

File: docs/vcs.rst [hunk 3]
```diff
@@ -1,17 +1,20 @@
 VCS Integration
 ===============
 
+If you want to integrate editing into your own theme, you will have to declare few variables inside your configuration file (config.py) in the "html_context" object, for the template to use them.
```
Comment (berkerpeksag) [inline docs/vcs.rst]: Please wrap all lines at 80 chars.

Comment (berkerpeksag) [inline docs/vcs.rst]: \`\`config.py\`\` -- also should the file name be `conf.py` instead?

Comment (berkerpeksag) [inline docs/vcs.rst]: "html_context" object -> \`\`'html_context'\`\` setting

We could also add a link to http://www.sphinx-doc.org/en/1.5.2/config.html#confval-html_context

File: docs/vcs.rst [hunk 4]
```diff
@@ -1,17 +1,20 @@
 VCS Integration
 ===============
 
+If you want to integrate editing into your own theme, you will have to declare few variables inside your configuration file (config.py) in the "html_context" object, for the template to use them. 
+
 GitHub
 ------
 
-If you want to integrate GitHub editing into your own theme, the following variables are available in your custom templates:
+Conf.py::
```
Comment (berkerpeksag) [inline docs/vcs.rst]: Perhaps say something like

```rst
If you want to integrate $SITE_NAME, you can put the following snippet into your ``conf.py``::
```

File: docs/vcs.rst [hunk 5]
```diff
@@ -1,17 +1,20 @@
...
 
-* github_user - GitHub username
-* github_repo - GitHub repo name
-* github_version - GitHub blob
-* conf_py_path - Path in the checkout to the docs root
-* pagename - Sphinx variable representing the name of the page you're on.
-* display_github
+      html_context = {
+        "display_github": True, # Integrate GitHub
```
Comment (berkerpeksag) [inline docs/vcs.rst]: 4 space indentation. e.g.

```py
html_context = {
    "display_github": True, # Integrate GitHub
}
```

File: docs/vcs.rst [hunk 6]
```diff
@@ -23,14 +26,17 @@ It can be used like this::
...
-If you want to integrate Bitbucket editing into your own theme, the following variables are available in your custom templates:
+If you want to integrate Bitbucket, use the following variables:
 
-* bitbucket_user - Bitbucket username
-* bitbucket_repo - Bitbucket repo name
-* bitbucket_version - BitBucket version
-* conf_py_path - Path in the checkout to the docs root
-* pagename - Sphinx variable representing the name of the page you're on.
-* display_bitbucket
+Conf.py::
```
Comment (berkerpeksag) [inline docs/vcs.rst]: Same as https://github.com/rtfd/readthedocs.org/pull/2592/files#r97964349

**Comment by humitos** [comment]
Is this PR dead? Is someone still working on this? It was reviewed a month ago and the requested changes seems to be easy to do.

Should I do those changes and propose a new review?

**Comment by naster01** [comment]
Thanks for the heads-up. 

Ideally the template [sphinx_rtd_theme](https://github.com/snide/sphinx_rtd_theme) would use generics variables for the integration with more VCS. Something like : 

- display_github -> display_cvs
- github_user -> cvs_user
- github_repo -> cvs_repo
- new -> cvs_logo

But this is for an other GitHub repo.

**Comment by berkerpeksag** [comment]
Thanks!

**Pull request merged**